### DISCUSSION TITLE
Use SDWire with OracleVirtualBox

### DISCUSSION THREAD

**Discussion by Petros626** [Dasharo/docs#469] [Q&A]
Oracle VirtualBox: 7.0.4.154605
OS: Ubuntu 20.04

Hello,

after i followed the guide for the SDWire I stuck at point: (`Unable to open ftdi device: -7 (set baudrate failed)`) . The showed solution is to set the configuration, which you usually find out with `dmesg -w`, but after inserting the specific values I get the SDWire is plugged out it's not possible to set:

`SerialNumber: 000000264001`
`idVendor=0x0424`
`idProduct=0x4050`

`sudo sd-mux-ctrl --device-serial=000000264001 --vendor=0x0424 --product=0x4050 --device-type=sd-wire --set-serial=sd-wire_11`

and logically after that the device can't be found (`Number of FTDI devices found: 0`) anymore. Would appreciate it, if someone could help me with this.

Thanks

**Comment by Pokisiekk** [comment]
Hi @Petros626

I was trying to reproduce your issue by installing the same version of VirtualBox and Ubuntu as you provided, but all SDWires, which I tested works form me. If it possible, try check it on real system or at least Ubuntu 22.04 as a virtual machine to rule out a hardware issue.

Let me know what it looks like and I will try to help further. If the problem is reproducible on other systems, I will ask for output from `dmesg` command (snippet about SDWire is enough).

**Reply by Petros626** [reply]
Hello @Pokisiekk 

thanks für answering. 

The SDWire on a real based Ubuntu 20.04 works fine without any problem. Only in the VM (host Windows10) the problems I said appear. I'll send you on Tuesday a Screenshot of the output with the command `dmesg -w` so you can reproduce maybe the issue.

Stay tuned and thanks again

**Reply by Petros626** [reply]
Hello @Pokisiekk

the attached screenshot here when I mount the `SDWire` to my PC (Attention I use a USB hub). By using the normal USB ports of the PC it laso doesn't work.

![image](https://user-images.githubusercontent.com/62354721/220257245-60390951-2fed-4e7b-b2a4-618c1c53a80f.png)

**Reply by Petros626** [reply]
@Pokisiekk for me the output of the attached device seems fine. The problem is, that as I showed in the answer (below) the access to the device

**Reply by Pokisiekk** [reply]
I've never encountered this problem(`ftdi_usb_get_strings failed: -4 (libusb_open() failed)`), but I can give you some tips:

1. Check that the `libusb` library is installed on your system.
2. Make sure that you have permission to access the device. You may need to be logged in as root or use `sudo` to run the program that is trying to access the device.
3. Try unplugging and re-plugging the device. Sometimes this can help reset the connection and resolve any issues.

I apologize for the late response.

**Comment by sulewskiprzemyslaw** [comment]
Hello @Petros626,

the values for idVendor and idProduct look a little bit weird. Could you confirm to me that you obtained them by using the `dmesg` command?

Also second question: Would you mind trying the following command?

```
sd-mux-ctrl --list --vendor=0x0424 --product=0x4050
```

This command comes from the sd-mux-ctrl manual and, according to the developers, it should be used in situations other than standard idVendor and idProduct.

**Reply by Petros626** [reply]
Hello @sulewskiprzemyslaw 

thank you also for answering this question. 

As I mentioned on Monday Tuesday I'll show the terminal output additionally with the suggested command.

**Reply by Petros626** [reply]
Hey @sulewskiprzemyslaw ,

`dmesg -w`:

![image](https://user-images.githubusercontent.com/62354721/220258545-8f56975e-7884-4bf4-b96f-036cfc244c4b.png)

I tired the following command, but it was the origin of my problem with the VM, that somehow the` ftdi library` failed to open.

![image](https://user-images.githubusercontent.com/62354721/220257755-f57b9365-46d2-4ef9-8a79-962d9b7a3403.png)

**Reply by Petros626** [reply]
any hints how to solve? @sulewskiprzemyslaw
